src/main/java/com/pubnub/api/workers/SubscribeMessageWorker.java

```diff
@@ -193,7 +195,13 @@ private void processIncomingPayload(SubscribeMessage message) {
                     .userMetadata(message.getUserMetadata())
                     .build();
 
-            listenerManager.announce(pnMessageResult, message.getType());
+            if (message.getType() == null) {
+                listenerManager.announce(new PNMessageResult(messageResult));
```

Comment:
Should this be `new MessageResult`?

Reply:
@davidnub 
If the `e` property is empty, it's a regular message, just like before Signals.